File: ts/src/test/base/test.isJsonEncodedObject.ts



import assert from 'assert';
import ccxt from '../../../ccxt.js';

function testIsJsonEncodedObject () {
    const exchange = new ccxt.Exchange ({
        'id': 'sampleexchange',
    });
    assert (exchange.isJsonEncodedObject ('{"key":"value"}'));
    assert (exchange.isJsonEncodedObject ('{}'));
    assert (exchange.isJsonEncodedObject ('[]'));
    // todo: the belows  are not ideal, but currently valid
    assert (exchange.isJsonEncodedObject ('{x'));
    assert (exchange.isJsonEncodedObject ('[x'));
    assert (exchange.isJsonEncodedObject ('{'));
    assert (exchange.isJsonEncodedObject ('['));
    // invalid
    assert (!exchange.isJsonEncodedObject ('x'));
    assert (!exchange.isJsonEncodedObject (''));
    assert (!exchange.isJsonEncodedObject ('}'));
    assert (!exchange.isJsonEncodedObject (']'));
    assert (!exchange.isJsonEncodedObject ('null'));
    assert (!exchange.isJsonEncodedObject ('undefined'));
}

export default testIsJsonEncodedObject;

File: julia/Ccxt/test/base/test.isJsonEncodedObject.jl
using Test
using Ccxt
function testIsJsonEncodedObject()

    exchange = get(ccxt, Symbol("Exchange"), nothing)(Dict{Symbol, Any}(
        Symbol("id") => "sampleexchange"
    ));
    @test functions.ccxtruthy(isJsonEncodedObject(exchange, "{\"key\":\"value\"}"));
    @test functions.ccxtruthy(isJsonEncodedObject(exchange, "{}"));
    @test functions.ccxtruthy(isJsonEncodedObject(exchange, "[]"));
    @test functions.ccxtruthy(isJsonEncodedObject(exchange, "{x"));
    @test functions.ccxtruthy(isJsonEncodedObject(exchange, "[x"));
    @test functions.ccxtruthy(isJsonEncodedObject(exchange, "{"));
    @test functions.ccxtruthy(isJsonEncodedObject(exchange, "["));
    @test !functions.ccxtruthy(isJsonEncodedObject(exchange, "x"));
    @test !functions.ccxtruthy(isJsonEncodedObject(exchange, ""));
    @test !functions.ccxtruthy(isJsonEncodedObject(exchange, "}"));
    @test !functions.ccxtruthy(isJsonEncodedObject(exchange, "]"));
    @test !functions.ccxtruthy(isJsonEncodedObject(exchange, "null"));
    @test !functions.ccxtruthy(isJsonEncodedObject(exchange, "undefined"));
end
